frontend/src/components/SamplesList.tsx

```diff
@@ -229,6 +230,7 @@ export default function SamplesList({
     }
 
     setUnsavedChanges?.(true);
+    gridRef.current?.api?.refreshCells({ rowNodes: [rowNode] });
```

Comment:
I noticed that cells are not "yellow" after being updated the first time, only the second time.

The problem is that this `unsubmittedChange` styling class isn't applied immediately after a cell value changes because the state update is asynchronous, and AG Grid might re-render the cell before the state is updated.

These lines force AG Grid to re-render the row right after updating the state, ensuring the class rule is applied correctly.